File: frontend/lib/models/area.dart
class Area {
  final int id;
  final String nombre;
  final String? codigo;
  final String? descripcion;
  final bool activo;

  Area({
    required this.id,
    required this.nombre,
    this.codigo,
    this.descripcion,
    required this.activo,
  });

  factory Area.fromJson(Map<String, dynamic> json) {
    return Area(
      id: json['id'],
      nombre: json['nombre'],
      codigo: json['codigo'],
      descripcion: json['descripcion'],
      activo: json['activo'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nombre': nombre,
      'codigo': codigo,
      'descripcion': descripcion,
      'activo': activo,
    };
  }
}

class CreateAreaDTO {
  final String nombre;
  final String? codigo;
  final String? descripcion;
  final bool activo;

  CreateAreaDTO({
    required this.nombre,
    this.codigo,
    this.descripcion,
    this.activo = true,
  });

  Map<String, dynamic> toJson() {
    return {
      'nombre': nombre,
      'codigo': codigo,
      'descripcion': descripcion,
      'activo': activo,
    };
  }
}

File: backend/Models/Area.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SistemaGestionDocumental.Models;

[Table("areas")]
public class Area
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Required]
    [Column("nombre")]
    [StringLength(100)]
    public string Nombre { get; set; } = string.Empty;

    [Column("codigo")]
    [StringLength(20)]
    public string? Codigo { get; set; }

    [Column("descripcion")]
    [StringLength(300)]
    public string? Descripcion { get; set; }

    [Column("activo")]
    public bool Activo { get; set; } = true;

    // Relaciones
    public virtual ICollection<Documento> DocumentosOrigen { get; set; } = new List<Documento>();
    public virtual ICollection<Movimiento> MovimientosOrigen { get; set; } = new List<Movimiento>();
    public virtual ICollection<Movimiento> MovimientosDestino { get; set; } = new List<Movimiento>();
    public virtual ICollection<Usuario> Usuarios { get; set; } = new List<Usuario>();
}

